var/spack/repos/builtin/packages/squashfs/package.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+class Squashfs(MakefilePackage):
+    """Squashfs - read only compressed filesystem"""
+
+    homepage = 'http://squashfs.sourceforge.net'
+    url      = 'https://downloads.sourceforge.net/project/squashfs/squashfs/squashfs4.3/squashfs4.3.tar.gz'
+
+    # version      md5sum
```

Comment:
Nowadays we prefer sha256 over md5. At some point we plan on updating all the older packages, but for new packages we prefer to use sha256.

Reply:
Is sha1 okay? Looks like squashfs provides that along with md5:

![image](https://user-images.githubusercontent.com/814322/55370400-610dd680-54c8-11e9-91f8-69b15f511324.png)

Reply:
sha1 is certainly better than md5. You can also download, verify the sha1/md5, then compute the sha256.

On macOS:
```console
$ shasum -a 256 squashfs4.3.tar.gz
```
On Linux:
```console
$ sha256sum squash4.3.tar.gz
```
If it was any other website than SourceForge, you could run `spack checksum` to do all of this automatically, but in my experience SourceForge prevents web crawlers.

Reply:
Looks like it's ok?

```python
#: Set of hash algorithms that Spack can use, mapped to digest size in bytes
hashes = {
    'md5': 16,
    'sha1': 20,
    'sha224': 28,
    'sha256': 32,
    'sha384': 48,
    'sha512': 64
}
```

Reply:
Yep, Spack can handle any hashing algorithm, although sha256 is the current default when you create a new package/add a new version.

Reply:
okay all set! I just pushed sha1, and I'm running locally to test again.